tasm-lib/src/test_helpers.rs

```diff
@@ -414,6 +415,11 @@ pub fn test_rust_equivalence_given_complete_state<T: RustShadow>(
     sponge: &Option<VmHasher>,
     expected_final_stack: Option<&[BFieldElement]>,
 ) -> VMState {
+    shadowed_snippet
+        .inner()
+        .borrow()
+        .assert_all_sign_offs_are_up_to_date();
+
```

Comment:
I'm unsure about this addition because it lives in a publicly exposed function. Should downstream callers also be exposed to the potential panics generated here?